// SPDX-License-Identifier: BSD-3-Clause
// SPDX-FileCopyrightText: 2018-2020 Carles Fernandez-Prades <carles.fernandez@cttc.es>
syntax = "proto3";

package gnss_sdr;

message MonitorGpsEphemeris {
    uint32 i_satellite_PRN=1;  // SV PRN NUMBER
    double d_TOW=2;             //!< Time of GPS Week of the ephemeris set (taken from subframes TOW) [s]
    double d_Crs=3;             //!< Amplitude of the Sine Harmonic Correction Term to the Orbit Radius [m]
    double d_Delta_n=4;         //!< Mean Motion Difference From Computed Value [semi-circles/s]
    double d_M_0=5;             //!< Mean Anomaly at Reference Time [semi-circles]
    double d_Cuc=6;             //!< Amplitude of the Cosine Harmonic Correction Term to the Argument of Latitude [rad]
    double d_e_eccentricity=7;  //!< Eccentricity [dimensionless]
    double d_Cus=8;             //!< Amplitude of the Sine Harmonic Correction Term to the Argument of Latitude [rad]
    double d_sqrt_A=9;          //!< Square Root of the Semi-Major Axis [sqrt(m)]
    uint32 d_Toe=10;             //!< Ephemeris data reference time of week (Ref. 20.3.3.4.3 IS-GPS-200K) [s]
    uint32 d_Toc=11;             //!< clock data reference time (Ref. 20.3.3.3.3.1 IS-GPS-200K) [s]
    double d_Cic=12;           //!< Amplitude of the Cosine Harmonic Correction Term to the Angle of Inclination [rad]
    double d_OMEGA0=13;        //!< Longitude of Ascending Node of Orbit Plane at Weekly Epoch [semi-circles]
    double d_Cis=14;           //!< Amplitude of the Sine Harmonic Correction Term to the Angle of Inclination [rad]
    double d_i_0=15;           //!< Inclination Angle at Reference Time [semi-circles]
    double d_Crc=16;           //!< Amplitude of the Cosine Harmonic Correction Term to the Orbit Radius [m]
    double d_OMEGA=17;         //!< Argument of Perigee [semi-cicles]
    double d_OMEGA_DOT=18;     //!< Rate of Right Ascension [semi-circles/s]
    double d_IDOT=19;          //!< Rate of Inclination Angle [semi-circles/s]
    int32 i_code_on_L2=20;   //!< If 1, P code ON in L2;  if 2, C/A code ON in L2;
    int32 i_GPS_week=21;     //!< GPS week number, aka WN [week]
    bool b_L2_P_data_flag=22;  //!< When true, indicates that the NAV data stream was commanded OFF on the P-code of the L2 channel
    int32 i_SV_accuracy=23;  //!< User Range Accuracy (URA) index of the SV (reference paragraph 6.2.1) for the standard positioning service user (Ref 20.3.3.3.1.3 IS-GPS-200K)
    int32 i_SV_health=24;
    double d_TGD=25;       //!< Estimated Group Delay Differential: L1-L2 correction term only for the benefit of "L1 P(Y)" or "L2 P(Y)" s users [s]
    double d_IODC=26;      //!< Issue of Data, Clock
    double d_IODE_SF2=27;  //!< Issue of Data, Ephemeris (IODE), subframe 2
    double d_IODE_SF3=28;  //!< Issue of Data, Ephemeris(IODE), subframe 3
    int32 i_AODO=29;     //!< Age of Data Offset (AODO) term for the navigation message correction table (NMCT) contained in subframe 4 (reference paragraph 20.3.3.5.1.9) [s]

    bool b_fit_interval_flag=30;  //!< indicates the curve-fit interval used by the CS (Block II/IIA/IIR/IIR-M/IIF) and SS (Block IIIA) in determining the ephemeris parameters, as follows: 0 = 4 hours, 1 = greater than 4 hours.
    double d_spare1=31;
    double d_spare2=32;

    double d_A_f0=33;  //!< Coefficient 0 of code phase offset model [s]
    double d_A_f1=34;  //!< Coefficient 1 of code phase offset model [s/s]
    double d_A_f2=35;  //!< Coefficient 2 of code phase offset model [s/s^2]
    
    bool b_integrity_status_flag=36;
    bool b_alert_flag=37;         //!< If true, indicates  that the SV URA may be worse than indicated in d_SV_accuracy, use that SV at our own risk.
    bool b_antispoofing_flag=38;  //!<  If true, the AntiSpoofing mode is ON in that SV
}
